unit_tests/log_transfer/test_main_chkpt_info.cpp

```diff
@@ -0,0 +1,800 @@
+  unpac.set_buffer (buffer, size);
+  env.after.unpack (unpac);
+
+  env.require_equal (env.before, env.after);
+}
+
+TEST_CASE ("Test pack/unpack checkpoint_info class 2", "")
+{
+  test_env_chkpt env {0, 100};
+
+  cubpacking::packer pac;
+  size_t size = env.before.get_packed_size (pac, 0);
+  char buffer[size];
+  pac.set_buffer (buffer, size);
+  env.before.pack (pac);
+
+  cubpacking::unpacker unpac;
+  unpac.set_buffer (buffer, size);
+  env.after.unpack (unpac);
+
+  env.require_equal (env.before, env.after);
+}
+
+TEST_CASE ("Test pack/unpack checkpoint_info class 3", "")
+{
+  test_env_chkpt env {220, 80};
+
+  cubpacking::packer pac;
+  size_t size = env.before.get_packed_size (pac, 0);
+  size += env.before.get_packed_size (pac, size);
+  char buffer[size];
+  pac.set_buffer (buffer, size);
+  env.before.pack (pac);
+  env.before.pack (pac);
+
+  cubpacking::unpacker unpac;
+  unpac.set_buffer (buffer, size);
+  env.after.unpack (unpac);
+  env.require_equal (env.before, env.after);
+
+  checkpoint_info after_2;
+
+  after_2.unpack (unpac);
+
+  env.require_equal (env.after, after_2);
+
+}
+
+TEST_CASE ("Test load and recovery on empty tran table", "")
+{
+  test_env_chkpt env;
+  LOG_LSA smallest_lsa;
+  LOG_LSA star_lsa;
+  THREAD_ENTRY thd;
+
+  env.after.load_trantable_snapshot (&thd, smallest_lsa);
+  env.after.recovery_analysis (&thd, star_lsa);
+  env.after.recovery_2pc_analysis (&thd);
+
+}
+
+std::map<TRANID, log_tdes *> tran_map;
+int afterCount;
+
+int
+search_for_id (int id)
+{
+  for (int i = 0; i < log_Gl.trantable.num_total_indices; i++)
+    {
+      if (log_Gl.trantable.all_tdes[i]->trid == id)
+	{
+	  return 1;
+	}
+    }
+  return 0;
+}
+
+void
+fullCompareTDES (log_tdes *td1, log_tdes *td2)
+{
+  REQUIRE (td1->head_lsa == td2->head_lsa);
+  REQUIRE (td1->tail_lsa == td2->tail_lsa);
+  REQUIRE (td1->undo_nxlsa == td2->undo_nxlsa);
+  REQUIRE (td1->posp_nxlsa == td2->posp_nxlsa);
+  REQUIRE (td1->savept_lsa == td2->savept_lsa);
+  REQUIRE (td1->tail_topresult_lsa == td2->tail_topresult_lsa);
+}
+
+void
+check_recovery (checkpoint_info obj)
+{
+  std::map<TRANID, log_tdes *>::iterator itr;
+  for (itr = tran_map.begin(); itr != tran_map.end(); ++itr)
+    {
+      REQUIRE (itr->first != NULL_TRANID);
+    }
+
+  for (int i = 1; i < log_Gl.trantable.num_total_indices; i++)
+    {
+      LOG_TDES *tdes = log_Gl.trantable.all_tdes[i];
+      std::map<TRANID, log_tdes *>::const_iterator tdes_after = tran_map.find (tdes->trid);
+
+      if (tdes->trid != NULL && tdes->tail_lsa == NULL_LSA)
+	{
+	  REQUIRE (tdes_after == tran_map.end());
+	  continue;
+	}
+      REQUIRE (tdes_after != tran_map.end());
+      if (tdes->state == TRAN_ACTIVE || tdes->state == TRAN_UNACTIVE_ABORTED)
+	{
+	  REQUIRE (tdes_after->second->state == TRAN_UNACTIVE_UNILATERALLY_ABORTED);
+	  continue;
+	}
+      REQUIRE (tdes_after->second->state == tdes->state);
+      fullCompareTDES (tdes, tdes_after->second);
+    }
+}
+int
```

Comment:
- Empty line between function definitions
- size_t for count

```suggestion
}

size_t
```